src/suggest-ladder.ts

```diff
@@ -16,32 +20,33 @@ export default async function suggestLadder(
   filterFunction: (bitrate: number, resolution: Resolution, vmaf: number) => boolean = () => true,
   includeAllBitrates: boolean = false,
   onProgress: (index: number, filename: string, total: number) => void = () => {}
-): Promise<BitrateResolutionVMAF[]> {
-  let pairs = new Map<number, { resolution: Resolution; vmaf: number }[]>();
+): Promise<LadderAndVmafPairs> {
+  const pairs = await pairVmafWithResolutionAndBitrate(directoryWithVmafFiles, filterFunction, onProgress);
+  // let pairs = new Map<number, { resolution: Resolution; vmaf: number }[]>();
   let optimal: { resolution: Resolution; vmaf: number; bitrate: number }[] = [];
 
-  logger.info('Loading VMAF data...');
-  const vmafs = await getVmaf(directoryWithVmafFiles, onProgress);
-  let counter = 1;
-  vmafs.forEach(({ filename, vmaf }) => {
-    const [resolutionStr, bitrateStr] = filename.split('_');
-    const [widthStr, heightStr] = resolutionStr.split('x');
+  // logger.info('Loading VMAF data...');
+  // const vmafs = await getVmaf(directoryWithVmafFiles, onProgress);
+  // let counter = 1;
+  // vmafs.forEach(({ filename, vmaf }) => {
+  //   const [resolutionStr, bitrateStr] = filename.split('_');
+  //   const [widthStr, heightStr] = resolutionStr.split('x');
 
-    const width = parseInt(widthStr);
-    const height = parseInt(heightStr);
-    const bitrate = parseInt(bitrateStr);
+  //   const width = parseInt(widthStr);
+  //   const height = parseInt(heightStr);
+  //   const bitrate = parseInt(bitrateStr);
 
-    if (filterFunction(bitrate, { width, height }, vmaf)) {
-      if (pairs.has(bitrate)) {
-        pairs.get(bitrate)?.push({ resolution: { width, height }, vmaf });
-      } else {
-        pairs.set(bitrate, [{ resolution: { width, height }, vmaf }]);
-      }
-    }
+  //   if (filterFunction(bitrate, { width, height }, vmaf)) {
+  //     if (pairs.has(bitrate)) {
+  //       pairs.get(bitrate)?.push({ resolution: { width, height }, vmaf });
+  //     } else {
+  //       pairs.set(bitrate, [{ resolution: { width, height }, vmaf }]);
+  //     }
+  //   }
 
-    logger.info(`Finished loading VMAF ${counter}/${vmafs.length}.`);
-    counter += 1;
-  });
+  //   logger.info(`Finished loading VMAF ${counter}/${vmafs.length}.`);
+  //   counter += 1;
+  // });
```

Comment:
Same as above. Maybe better to remove it than leave dead code?

Reply:
fixed